kbfsedits/user_history.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package kbfsedits
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/tlf"
+)
+
+const (
+	// The max number of TLF writer clusters to return in a user history.
+	maxClusters = 10
+)
+
+type tlfKey struct {
+	tlfName tlf.CanonicalName
```

Comment:
In case of SBS, will we end up with two `tlfWithHistory` objects?

Reply:
Well, `folderBranchOps.monitorEditsChat` only gets the name once at the beginning of its loop, and uses that same name every time it calls `UpdateHistory`.  So no, I don't see how there would be two `TlfHistory` objects for the same TLF, but the existing one will continue to have the old name I think, even after the assertion is proven.  The next KBFS restart will fix it.